modulemd/modulemd-module-stream.c

```diff
@@ -942,12 +948,53 @@ stream_expansion_helper (ModulemdDependencies *deps,
 
       /*
        * If a module is present in the dependency list but has no associated
-       * streams (which corresponds to "modulename: []" in the spec), the
-       * intention is to expand the list to be all active existing streams for
-       * the module. Unfortunately, that is something only the Module Build
-       * Service can do, so we must fail the stream expansion.
+       * streams (which corresponds to "modulename: []" in the spec), or it
+       * contains only negated items (corresponding to
+       * "modulename: [-stream]"), the intention is to expand the list to be
+       * all active existing streams for the module. Check to see if we have a
+       * listing of known streams to use. We can assume if the first item
+       * begins with a '-' that they all do, since that is checked during the
+       * parsing and validation of the YAML.
        */
-      if (g_strv_length (streams) == 0)
+      if (g_strv_length (streams) == 0 || streams[0][0] == '-')
+        {
+          if (helper)
+            {
+              known_streams =
+                modulemd_upgrade_helper_get_known_streams_as_array (helper,
+                                                                    module);
+
+              /* Check whether we got back at least one stream */
+              if (known_streams->len > 0)
+                {
+                  /* Iterate through the negations */
+                  for (j = 0; streams[j] == NULL; j++)
+                    {
+                      /* Skip the leading '-' */
+                      stream = streams[j] + 1;
+
+                      /* Remove it if it exists in the list */
+                      g_ptr_array_remove (known_streams, stream);
+                    }
+
+
+                  /* Add NULL-terminator */
+                  g_ptr_array_add (known_streams, NULL);
+
+                  /* Replace the streams list with the known list */
+                  g_clear_pointer (&streams, g_strfreev);
+                  streams = (GStrv)known_streams->pdata;
+                }
+
+              /* Make sure to free the array, but not the contents */
+              g_ptr_array_free (known_streams, FALSE);
```

Comment:
Could you please add a debugging message here something along the lines of:
```
      g_debug ("Expansion: module %s dependency %s now has %d streams after helper",
               which,
               module,
               g_strv_length (streams));
```

Reply:
Will do